### PR TITLE
✨ Add `BusinessCore` source and Prefect tasks 

### PR DESCRIPTION
**Pull request by judynah**
## Summary
Business Core connector migration from viadot 1 to viadot 2. Added `verify` parameter into utility functions.

## Importance
Required by the migration project.

## Checklist

This PR:
- [x] follows the guidelines laid out in `CONTRIBUTING.md`
- [ ] links relevant issue(s)
- [x] adds/updates tests (if appropriate)
- [x] adds/updates docstrings (if appropriate)
- [ ] adds an entry in `CHANGELOG.md`

### CODE REVIEW

**Comment by trymzet** [comment]
@judynah pls fix the CI errors before requesting review

**Review by trymzet — CHANGES_REQUESTED** [review]
CI not passing so not ready for review

File: src/viadot/sources/business_core.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        Raises:
+            CredentialError: When credentials are not found.
+        """
+        raw_creds = credentials or get_source_credentials(config_key) or None
+        error_message = "Missing Credentials."
+        if raw_creds is None:
+            raise CredentialError(error_message)
+
+        validated_creds = dict(BusinessCoreCredentials(**raw_creds))
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:67]:
```suggestion
        raw_creds = credentials or get_source_credentials(config_key)
        if raw_creds is None:
            msg = "Missing Credentials."
            raise CredentialError(msg)
        validated_creds = dict(BusinessCoreCredentials(**raw_creds))
```

File: src/viadot/sources/business_core.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        - username: The unique username for the organization.
+        - password: A secret string of characters for access to data.
+
+    Args:
+        BaseModel (pydantic.main.ModelMetaclass): A base class for creating
+            Pydantic models.
+    """
+
+    username: str
+    password: str
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:28]:
```suggestion
    password: SecretStr
```

File: src/viadot/sources/business_core.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    def generate_token(self) -> str:
+        """Function for generating Business Core token based on username and password.
+
+        Returns:
+            string: Business Core API token.
+        """
+        url = "https://api.businesscore.ae/api/user/Login"
+
+        payload = f'grant_type=password&username={self.credentials.get("username")}&password={self.credentials.get("password")}&scope='
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:84]:
```suggestion
        username = self.credentials.get("username")
        password = self.credentials.get("password").get_secret_value()
        payload = f'grant_type=password&username={username}&password={password}&scope='
```

File: src/viadot/sources/business_core.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        response = handle_api_response(
+            url=url,
+            headers=headers,
+            method="GET",
+            data=payload,
+            verify=self.verify,
+        )
+        token = json.loads(response.text).get("access_token")
+        self.token = token
+        return token
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:95]:
```suggestion
        return json.loads(response.text).get("access_token")
```

File: src/viadot/sources/business_core.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            Needed for payload in 'x-www-form-urlencoded' from.
+
+        Returns:
+            Dict: Dictionary with filters prepared for further use.
+        """
+        return {
+            key: ("&" if val is None else val) for key, val in self.filters_dict.items()
+        }
+
+    def get_data(self) -> dict:
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:109]: Should be refactored into `to_records()`; see existing issues regarding `api_connection()`, as this is the same case here. For now pls create an issue and link it here.

File: src/viadot/sources/business_core.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+            "GetPendingSalesOrderData",
+            "GetSalesInvoiceData",
+            "GetSalesReturnDetailData",
+            "GetSalesOrderData",
+            "GetSalesQuotationData",
+        ]:
+            error_message = f"View {view} currently not available."
+            raise APIError(error_message)
+
+        data = self.get_data().get("MasterDataList")
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:169]: `get_data()`/`to_records()` should already return the data we're interested in, so this filtering should be done there

File: src/viadot/sources/business_core.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+            "GetCustomerData",
+            "GetItemMaster",
+            "GetPendingSalesOrderData",
+            "GetSalesInvoiceData",
+            "GetSalesReturnDetailData",
+            "GetSalesOrderData",
+            "GetSalesQuotationData",
+        ]:
+            error_message = f"View {view} currently not available."
+            raise APIError(error_message)
```
Comment (trymzet) [inline src/viadot/sources/business_core.py:167]: Shouldn't this be handled in `get_data()`?

**Review by trymzet — CHANGES_REQUESTED** [review]
Added my feedback

File: src/viadot/sources/business_core.py (hunk 5 above)
Comment (judynah) [reply src/viadot/sources/business_core.py:107]: created issue: https://github.com/dyvenia/viadot/issues/1078

**Review by trymzet — APPROVED** [review]

**Pull request merged**